text/MateuszPabian/main.tex

```diff
@@ -0,0 +1,186 @@
+\documentclass[a4paper,11pt]{article}
+\usepackage[utf8]{inputenc}
+\usepackage{polski}
+\usepackage{graphicx}
+\usepackage[left=1in,right=1in]{geometry}
+\usepackage{algpseudocode}
+\usepackage{algorithm}
+\usepackage{amsthm}
+
+\newgeometry{tmargin=2cm, bmargin=2cm, lmargin=2cm, rmargin=2cm}
+\renewcommand*{\proofname}{\textbf{Rozwiązanie}}
+
+\newtheorem{DEF}{\textsc{Definicja}}[section]
+\newtheorem{OBS}[DEF]{\textsc{Obserwacja}}
+\newtheorem{LEM}[DEF]{\textsc{Lemat}}
+\newtheorem{PROP}[DEF]{\textsc{Propozycja}}
+\newtheorem{TW}[DEF]{\textsc{Twierdzenie}}
+\newtheorem{HIP}[DEF]{\textsc{Hipoteza}}
+\newtheorem*{N}{\textsc{Notacja}}
+\newtheorem{U}{\textsc{Uwaga}}
+\newtheorem{W}{\textsc{Wniosek}}
+\newtheorem*{PROF}{\textsc{Dowód}}
+
+\title{\textbf{Fast practical multi-pattern matching}}
+\author{\textbf{Mateusz Pabian}}
+\date{}
+
+\begin{document}
+\maketitle
+
+\section{Wprowadzenie}
+\textit{Fast practical multi-pattern matching} to algorytm tekstowy służący do wyszukiwania w zadanym tekście wystąpień słów z podanego zbioru wzorców.
+
+Algorytm jest ulepszeniem algorytmu Commentz-Waltera, czerpiąc z niego kilka istotnych obserwacji takich jak przeskakiwanie nieinteresujących obszarów tekstu oraz wykorzystanie automatu Aho-Corasick. Dodatkowym elementem zwiększającym efektywność jest wykorzystanie automatu sufiksowego zwanego dalej \textit{DAWG}.   
+
+Algorytm, który jest tematem pracy, dla tekstu o długości $n$ ma pesymistyczny czas działania $O(n)$.
+W przypadku średnim, gdzie najkrótszy spośród wzorców ma długość $m$ a suma ich długości jest wielomianowa w zależności od $m$ oraz prawdopodobieństwo wystąpienia litery z alfabetu (conajmniej dwu-elementowego) jest jednostajne i niezależne, złożoność wynosi $O(\frac{n}{m}* \log{m})$, co jest równocześnie dolnym ograniczeniem dla tego problemu.
+Dodatkowo dla odpowienio dużego $m$ algorytm osiąga dobre rezultaty w praktyce.
+
+\section{Idea}
+
+Idea algorytmu oparta jest o dwie fazy zwane \textit{PROCESS1} oraz \textit{PROCESS2}. Pierwsza z nich skanuje test od lewej do prawej przesuwając potencjną pozycję zakończenia wzorca co najmniej o $\frac{m}{2}$. W tej fazie bazuje on na zmiennej $\gamma$, która spełnia niezmiennik $\gamma = $ najdłuży suffix od początku tekstu do pozycji $i$, który jest jednocześnie prefiksem jakiegoś wzorca. Jeśli długość $\gamma \leq \frac{m}{2}$ kończymy pierwszą fazę algorytmu i przechodzimy do fazy drugiej. W implementacji będziemy utorzsamiać $\gamma$ ze stanem w AC.
```

Comment:
Nit: please use some spellchecker, even in this paragraph I noticed "najdłuży" or "utorzsamiać".